app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -48,34 +72,121 @@ export default class FileAttachmentList extends Component {
         }
     }
 
-    goToImagePreview = (postId, fileId) => {
+    buildGalleryFiles = async (props) => {
+        const {files} = props;
+        const results = [];
+
+        if (files && files.length) {
+            for (let i = 0; i < files.length; i++) {
+                const file = files[i];
+                const caption = file.name;
+
+                if (isDocument(file) || isVideo(file) || (!file.has_preview_image && !isGif(file))) {
+                    results.push({
+                        caption,
+                        data: file,
+                    });
+                    continue;
+                }
+
+                let uri;
+                let cache;
+                if (file.localPath) {
+                    uri = file.localPath;
+                } else if (isGif(file)) {
+                    cache = await getCacheFile(file.name, Client4.getFileUrl(file.id));
+                } else {
+                    cache = await getCacheFile(file.name, Client4.getFilePreviewUrl(file.id));
+                }
+
+                if (cache) {
+                    let path = cache.path;
+                    if (Platform.OS === 'android') {
+                        path = `file://${path}`;
+                    }
+
+                    uri = path;
+                }
+
+                results.push({
+                    caption,
+                    source: {uri},
+                    data: file,
+                });
+            }
+        }
+
+        return results;
+    };
+
+    getItemMeasures = (index, cb) => {
+        const activeComponent = this.items[index];
+
+        if (!activeComponent) {
+            cb(null);
+            return;
+        }
+
+        activeComponent.measure((rx, ry, width, height, x, y) => {
+            cb({
+                origin: {x, y, width, height},
+            });
+        });
+    };
+
+    getPreviewProps = (index) => {
+        const previewComponent = this.previewItems[index];
+        return previewComponent ? {...previewComponent.props} : {};
```

Comment:
What is this used for? I don't think we're supposed to be reading props from a ref

Reply:
this is the easiest way I found to *clone* a component, and actually when cloning a component you do pass the props ;)